[core] Type inference error: updates count limit reached

### Issue details

The method fails to decompile, the simple mode also fails with the following:
<details>
<summary>Logs</summary>

```
JADX ERROR: Simple mode code generation failed
            jadx.core.utils.exceptions.CodegenException: Error generate insn: 0x0004: INSTANCE_OF (r2v0 ?? I:boolean) = (r33v0 l00.c) A[DECLARE_VAR] b9.s0 in method: b9.x0.a(int, l00.c, boolean):java.lang.Object, file: classes.dex
            	at jadx.core.codegen.InsnGen.makeInsn(InsnGen.java:310)
            	at jadx.core.codegen.InsnGen.makeInsn(InsnGen.java:273)
            	at jadx.core.codegen.MethodGen.generateSimpleCode(MethodGen.java:354)
            	at jadx.core.codegen.MethodGen.addSimpleMethodCode(MethodGen.java:322)
            	at jadx.core.codegen.MethodGen.addInstructions(MethodGen.java:285)
            	at jadx.core.codegen.ClassGen.addMethodCode(ClassGen.java:406)
            	at jadx.core.codegen.ClassGen.addMethod(ClassGen.java:335)
            	at jadx.core.codegen.ClassGen.lambda$addInnerClsAndMethods$3(ClassGen.java:301)
            	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
            	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
            	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:395)
            	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
            Caused by: java.lang.NullPointerException: Cannot invoke "jadx.core.dex.instructions.args.ArgType.getPrimitiveType()" because "type" is null
            	at jadx.core.codegen.ClassGen.useType(ClassGen.java:550)
            	at jadx.core.codegen.InsnGen.useType(InsnGen.java:269)
            	at jadx.core.codegen.InsnGen.declareVar(InsnGen.java:166)
            	at jadx.core.codegen.InsnGen.declareVar(InsnGen.java:159)
            	at jadx.core.codegen.InsnGen.assignVar(InsnGen.java:152)
            	at jadx.core.codegen.InsnGen.makeInsn(InsnGen.java:299)
            	... 11 more
```
</details>

### Relevant log output or stacktrace

```java
JADX ERROR: Types fix failed
jadx.core.utils.exceptions.JadxOverflowException: Type inference error: updates count limit reached
        at jadx.core.utils.ErrorsCounter.addError(ErrorsCounter.java:59)
        at jadx.core.utils.ErrorsCounter.error(ErrorsCounter.java:31)
       	at jadx.core.dex.attributes.nodes.NotificationAttrNode.addError(NotificationAttrNode.java:19)
        at jadx.core.dex.visitors.typeinference.FixTypesVisitor.visit(FixTypesVisitor.java:96)
```

### Provide sample and class/method full name

Smali of the affected method: [method_smali.txt](https://github.com/user-attachments/files/21435801/method_smali.txt)

Additionally, I compiled jadx with [`updatesLimitCount`](https://github.com/skylot/jadx/blob/b1f48f1db1191d2da43ce3b8ec43d938ea83594c/jadx-core/src/main/java/jadx/core/dex/visitors/typeinference/TypeUpdateInfo.java#L24) set to `mth.getInsnsCount() * 50`, which doesn't give the TypeInterference error and instead outputs the following:

<details>
<summary>Logs</summary>

```
    /* JADX WARN: Can't wrap try/catch for region: R(12:254|(2:468|255)|(11:257|472|258|259|262|495|263|264|452|265|(7:268|456|269|270|482|271|(4:274|474|275|(3:460|277|(4:280|448|281|(2:283|(4:286|446|287|(2:289|(1:291)(3:292|293|(7:296|480|297|298|462|299|(4:302|470|303|(3:494|305|(1:307)(3:308|309|(1:311)(3:312|313|(2:315|(1:317)(4:318|319|320|(2:322|(1:324)(9:325|326|476|340|341|342|373|(1:378)(1:377)|(2:380|(1:382)(3:383|384|(1:386)(2:387|(2:401|(1:403)(2:404|(1:406)(4:407|408|409|(1:411)(3:412|413|(1:415)(5:416|(2:418|(1:420))|421|431|432)))))(2:391|(1:393)(6:394|395|(1:397)|398|409|(0)(0))))))(2:422|(5:425|426|421|431|432))))(2:328|(1:330)(11:331|332|333|476|340|341|342|373|(3:375|378|(0)(0))(0)|431|432))))(10:334|335|476|340|341|342|373|(0)(0)|431|432))))(14:336|337|486|338|339|335|476|340|341|342|373|(0)(0)|431|432)))))(10:349|350|442|354|355|342|373|(0)(0)|431|432)))(10:353|350|442|354|355|342|373|(0)(0)|431|432)))(10:360|488|361|362|490|363|373|(0)(0)|431|432))))|260|261|262|495|263|264|452|265|(0)) */
    /* JADX WARN: Can't wrap try/catch for region: R(13:254|468|255|(11:257|472|258|259|262|495|263|264|452|265|(7:268|456|269|270|482|271|(4:274|474|275|(3:460|277|(4:280|448|281|(2:283|(4:286|446|287|(2:289|(1:291)(3:292|293|(7:296|480|297|298|462|299|(4:302|470|303|(3:494|305|(1:307)(3:308|309|(1:311)(3:312|313|(2:315|(1:317)(4:318|319|320|(2:322|(1:324)(9:325|326|476|340|341|342|373|(1:378)(1:377)|(2:380|(1:382)(3:383|384|(1:386)(2:387|(2:401|(1:403)(2:404|(1:406)(4:407|408|409|(1:411)(3:412|413|(1:415)(5:416|(2:418|(1:420))|421|431|432)))))(2:391|(1:393)(6:394|395|(1:397)|398|409|(0)(0))))))(2:422|(5:425|426|421|431|432))))(2:328|(1:330)(11:331|332|333|476|340|341|342|373|(3:375|378|(0)(0))(0)|431|432))))(10:334|335|476|340|341|342|373|(0)(0)|431|432))))(14:336|337|486|338|339|335|476|340|341|342|373|(0)(0)|431|432)))))(10:349|350|442|354|355|342|373|(0)(0)|431|432)))(10:353|350|442|354|355|342|373|(0)(0)|431|432)))(10:360|488|361|362|490|363|373|(0)(0)|431|432))))|260|261|262|495|263|264|452|265|(0)) */
    /* JADX WARN: Code restructure failed: missing block: B:195:0x070e, code lost:
    
        if (r0 == r11) goto L196;
     */
    /* JADX WARN: Code restructure failed: missing block: B:370:0x0b35, code lost:
    
        r27 = "VALID";
        r25 = r10;
        r33 = "EXPIRED";
        r7 = r14;
        r28 = r15;
        r10 = r23;
        r8 = r24;
        r0 = r0;
     */
    /* JADX WARN: Code restructure failed: missing block: B:371:0x0b45, code lost:
    
        r27 = "VALID";
        r25 = r10;
        r33 = "EXPIRED";
        r7 = r14;
        r28 = r15;
        r10 = r23;
        r8 = r24;
        r0 = r0;
     */
    /* JADX WARN: Multi-variable type inference failed */
    /* JADX WARN: Removed duplicated region for block: B:127:0x0580  */
    /* JADX WARN: Removed duplicated region for block: B:157:0x0645  */
    /* JADX WARN: Removed duplicated region for block: B:158:0x0646 A[Catch: Exception -> 0x0060, PHI: r3
      0x0646: PHI (r3v35 long) = (r3v33 long), (r3v36 long) binds: [B:156:0x0643, B:132:0x0591] A[DONT_GENERATE, DONT_INLINE], TryCatch #9 {Exception -> 0x0060, blocks: (B:13:0x0056, B:16:0x0067, B:19:0x0071, B:22:0x007f, B:25:0x0096, B:28:0x00a2, B:31:0x00b1, B:34:0x00c7, B:37:0x00db, B:40:0x00f7, B:90:0x046d, B:93:0x048b, B:96:0x04aa, B:97:0x04b8, B:98:0x04c1, B:99:0x04cc, B:102:0x04e0, B:105:0x04ee, B:108:0x04fb, B:111:0x050a, B:114:0x0520, B:117:0x0531, B:120:0x054b, B:123:0x0565, B:173:0x069b, B:124:0x0573, B:167:0x0682, B:126:0x057b, B:164:0x066c, B:129:0x0588, B:161:0x065a, B:132:0x0591, B:158:0x0646, B:135:0x05a4, B:155:0x0620, B:138:0x05b7, B:148:0x05fd, B:151:0x060b, B:169:0x0686, B:144:0x05d4), top: B:454:0x0042 }] */
    /* JADX WARN: Removed duplicated region for block: B:160:0x0659  */
    /* JADX WARN: Removed duplicated region for block: B:161:0x065a A[Catch: Exception -> 0x0060, PHI: r3
      0x065a: PHI (r3v37 long) = (r3v35 long), (r3v39 long) binds: [B:159:0x0657, B:129:0x0588] A[DONT_GENERATE, DONT_INLINE], TryCatch #9 {Exception -> 0x0060, blocks: (B:13:0x0056, B:16:0x0067, B:19:0x0071, B:22:0x007f, B:25:0x0096, B:28:0x00a2, B:31:0x00b1, B:34:0x00c7, B:37:0x00db, B:40:0x00f7, B:90:0x046d, B:93:0x048b, B:96:0x04aa, B:97:0x04b8, B:98:0x04c1, B:99:0x04cc, B:102:0x04e0, B:105:0x04ee, B:108:0x04fb, B:111:0x050a, B:114:0x0520, B:117:0x0531, B:120:0x054b, B:123:0x0565, B:173:0x069b, B:124:0x0573, B:167:0x0682, B:126:0x057b, B:164:0x066c, B:129:0x0588, B:161:0x065a, B:132:0x0591, B:158:0x0646, B:135:0x05a4, B:155:0x0620, B:138:0x05b7, B:148:0x05fd, B:151:0x060b, B:169:0x0686, B:144:0x05d4), top: B:454:0x0042 }] */
    /* JADX WARN: Removed duplicated region for block: B:163:0x066b  */
    /* JADX WARN: Removed duplicated region for block: B:166:0x0680  */
    /* JADX WARN: Removed duplicated region for block: B:169:0x0686 A[Catch: Exception -> 0x0060, TryCatch #9 {Exception -> 0x0060, blocks: (B:13:0x0056, B:16:0x0067, B:19:0x0071, B:22:0x007f, B:25:0x0096, B:28:0x00a2, B:31:0x00b1, B:34:0x00c7, B:37:0x00db, B:40:0x00f7, B:90:0x046d, B:93:0x048b, B:96:0x04aa, B:97:0x04b8, B:98:0x04c1, B:99:0x04cc, B:102:0x04e0, B:105:0x04ee, B:108:0x04fb, B:111:0x050a, B:114:0x0520, B:117:0x0531, B:120:0x054b, B:123:0x0565, B:173:0x069b, B:124:0x0573, B:167:0x0682, B:126:0x057b, B:164:0x066c, B:129:0x0588, B:161:0x065a, B:132:0x0591, B:158:0x0646, B:135:0x05a4, B:155:0x0620, B:138:0x05b7, B:148:0x05fd, B:151:0x060b, B:169:0x0686, B:144:0x05d4), top: B:454:0x0042 }] */
    /* JADX WARN: Removed duplicated region for block: B:171:0x0697  */
    /* JADX WARN: Removed duplicated region for block: B:172:0x0699  */
    /* JADX WARN: Removed duplicated region for block: B:179:0x06ae A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:184:0x06bc A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:189:0x06dc A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:194:0x06fe A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:199:0x0723 A[PHI: r3 r4 r22 r23 r24 r25
      0x0723: PHI (r3v48 int) = (r3v45 int), (r3v53 int) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]
      0x0723: PHI (r4v16 long) = (r4v14 long), (r4v17 long) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]
      0x0723: PHI (r22v16 java.lang.String) = (r22v13 java.lang.String), (r22v17 java.lang.String) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]
      0x0723: PHI (r23v12 java.lang.String) = (r23v10 java.lang.String), (r23v13 java.lang.String) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]
      0x0723: PHI (r24v9 java.lang.String) = (r24v6 java.lang.String), (r24v10 java.lang.String) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]
      0x0723: PHI (r25v25 java.lang.String) = (r25v23 java.lang.String), (r25v26 java.lang.String) binds: [B:188:0x06da, B:193:0x06fc] A[DONT_GENERATE, DONT_INLINE]] */
    /* JADX WARN: Removed duplicated region for block: B:201:0x0727 A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:207:0x0752  */
    /* JADX WARN: Removed duplicated region for block: B:210:0x0767  */
    /* JADX WARN: Removed duplicated region for block: B:211:0x0768  */
    /* JADX WARN: Removed duplicated region for block: B:214:0x077e  */
    /* JADX WARN: Removed duplicated region for block: B:216:0x0782 A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:222:0x07c1  */
    /* JADX WARN: Removed duplicated region for block: B:225:0x07d7  */
    /* JADX WARN: Removed duplicated region for block: B:228:0x07ee  */
    /* JADX WARN: Removed duplicated region for block: B:231:0x07f5 A[Catch: Exception -> 0x06b3, TryCatch #7 {Exception -> 0x06b3, blocks: (B:235:0x0809, B:229:0x07f0, B:226:0x07d9, B:223:0x07c3, B:220:0x079d, B:216:0x0782, B:212:0x076a, B:205:0x0748, B:208:0x0753, B:197:0x0711, B:192:0x06ef, B:194:0x06fe, B:201:0x0727, B:231:0x07f5, B:187:0x06d2, B:189:0x06dc, B:177:0x06a8, B:179:0x06ae, B:184:0x06bc), top: B:455:0x06a8 }] */
    /* JADX WARN: Removed duplicated region for block: B:239:0x081d  */
    /* JADX WARN: Removed duplicated region for block: B:240:0x081f  */
    /* JADX WARN: Removed duplicated region for block: B:243:0x0838  */
    /* JADX WARN: Removed duplicated region for block: B:244:0x083a  */
    /* JADX WARN: Removed duplicated region for block: B:247:0x083f  */
    /* JADX WARN: Removed duplicated region for block: B:254:0x085e  */
    /* JADX WARN: Removed duplicated region for block: B:257:0x0890 A[Catch: Exception -> 0x089b, TRY_LEAVE, TryCatch #15 {Exception -> 0x089b, blocks: (B:255:0x0864, B:257:0x0890), top: B:468:0x0864 }] */
    /* JADX WARN: Removed duplicated region for block: B:268:0x08c4  */
    /* JADX WARN: Removed duplicated region for block: B:274:0x08e7  */
    /* JADX WARN: Removed duplicated region for block: B:283:0x0912 A[Catch: Exception -> 0x0ab8, TRY_LEAVE, TryCatch #3 {Exception -> 0x0ab8, blocks: (B:281:0x090a, B:283:0x0912), top: B:448:0x090a }] */
    /* JADX WARN: Removed duplicated region for block: B:289:0x093a A[Catch: Exception -> 0x0aa9, TryCatch #2 {Exception -> 0x0aa9, blocks: (B:293:0x0951, B:287:0x0932, B:289:0x093a), top: B:446:0x0932 }] */
    /* JADX WARN: Removed duplicated region for block: B:296:0x0973  */
    /* JADX WARN: Removed duplicated region for block: B:302:0x0997  */
    /* JADX WARN: Removed duplicated region for block: B:311:0x09e6  */
    /* JADX WARN: Removed duplicated region for block: B:312:0x09e8  */
    /* JADX WARN: Removed duplicated region for block: B:315:0x09f7 A[Catch: Exception -> 0x0a48, TryCatch #28 {Exception -> 0x0a48, blocks: (B:332:0x0a66, B:326:0x0a37, B:319:0x0a12, B:322:0x0a1d, B:328:0x0a4d, B:313:0x09ed, B:315:0x09f7, B:309:0x09c8, B:305:0x09aa), top: B:494:0x09aa }] */
    /* JADX WARN: Removed duplicated region for block: B:322:0x0a1d A[Catch: Exception -> 0x0a48, TRY_ENTER, TryCatch #28 {Exception -> 0x0a48, blocks: (B:332:0x0a66, B:326:0x0a37, B:319:0x0a12, B:322:0x0a1d, B:328:0x0a4d, B:313:0x09ed, B:315:0x09f7, B:309:0x09c8, B:305:0x09aa), top: B:494:0x09aa }] */
    /* JADX WARN: Removed duplicated region for block: B:328:0x0a4d A[Catch: Exception -> 0x0a48, TryCatch #28 {Exception -> 0x0a48, blocks: (B:332:0x0a66, B:326:0x0a37, B:319:0x0a12, B:322:0x0a1d, B:328:0x0a4d, B:313:0x09ed, B:315:0x09f7, B:309:0x09c8, B:305:0x09aa), top: B:494:0x09aa }] */
    /* JADX WARN: Removed duplicated region for block: B:334:0x0a78  */
    /* JADX WARN: Removed duplicated region for block: B:336:0x0a7e A[Catch: Exception -> 0x0a9b, TRY_ENTER, TRY_LEAVE, TryCatch #16 {Exception -> 0x0a9b, blocks: (B:303:0x099e, B:336:0x0a7e), top: B:470:0x099e }] */
    /* JADX WARN: Removed duplicated region for block: B:349:0x0ab0  */
    /* JADX WARN: Removed duplicated region for block: B:353:0x0ac4  */
    /* JADX WARN: Removed duplicated region for block: B:360:0x0afc  */
    /* JADX WARN: Removed duplicated region for block: B:375:0x0b67  */
    /* JADX WARN: Removed duplicated region for block: B:378:0x0b6c  */
    /* JADX WARN: Removed duplicated region for block: B:380:0x0b6f A[Catch: Exception -> 0x0c02, TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:386:0x0bab  */
    /* JADX WARN: Removed duplicated region for block: B:387:0x0bad A[Catch: Exception -> 0x0c02, PHI: r1 r3 r6 r7 r25 r27 r28 r33
      0x0bad: PHI (r1v14 kotlin.jvm.internal.w) = (r1v13 kotlin.jvm.internal.w), (r1v112 kotlin.jvm.internal.w) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r3v10 long) = (r3v9 long), (r3v139 long) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r6v8 kotlin.jvm.internal.x) = (r6v7 kotlin.jvm.internal.x), (r6v67 kotlin.jvm.internal.x) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r7v10 b9.r0) = (r7v9 b9.r0), (r7v74 b9.r0) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r25v11 java.util.concurrent.atomic.AtomicBoolean) = (r25v10 java.util.concurrent.atomic.AtomicBoolean), (r25v86 java.util.concurrent.atomic.AtomicBoolean) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r27v9 java.lang.String) = (r27v8 java.lang.String), (r27v51 java.lang.String) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r28v9 bs.c) = (r28v8 bs.c), (r28v43 bs.c) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE]
      0x0bad: PHI (r33v11 java.lang.String) = (r33v10 java.lang.String), (r33v63 java.lang.String) binds: [B:385:0x0ba9, B:38:0x00de] A[DONT_GENERATE, DONT_INLINE], TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:397:0x0bfd  */
    /* JADX WARN: Removed duplicated region for block: B:401:0x0c05 A[Catch: Exception -> 0x0c02, TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:403:0x0c20  */
    /* JADX WARN: Removed duplicated region for block: B:404:0x0c22 A[Catch: Exception -> 0x0c02, PHI: r1 r3 r7 r25 r33
      0x0c22: PHI (r1v19 bs.c) = (r1v15 bs.c), (r1v113 bs.c) binds: [B:402:0x0c1e, B:29:0x00a5] A[DONT_GENERATE, DONT_INLINE]
      0x0c22: PHI (r3v13 long) = (r3v10 long), (r3v142 long) binds: [B:402:0x0c1e, B:29:0x00a5] A[DONT_GENERATE, DONT_INLINE]
      0x0c22: PHI (r7v13 b9.r0) = (r7v10 b9.r0), (r7v77 b9.r0) binds: [B:402:0x0c1e, B:29:0x00a5] A[DONT_GENERATE, DONT_INLINE]
      0x0c22: PHI (r25v14 java.util.concurrent.atomic.AtomicBoolean) = (r25v11 java.util.concurrent.atomic.AtomicBoolean), (r25v89 java.util.concurrent.atomic.AtomicBoolean) binds: [B:402:0x0c1e, B:29:0x00a5] A[DONT_GENERATE, DONT_INLINE]
      0x0c22: PHI (r33v12 java.lang.String) = (r33v11 java.lang.String), (r33v64 java.lang.String) binds: [B:402:0x0c1e, B:29:0x00a5] A[DONT_GENERATE, DONT_INLINE], TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:406:0x0c38  */
    /* JADX WARN: Removed duplicated region for block: B:407:0x0c3a  */
    /* JADX WARN: Removed duplicated region for block: B:411:0x0c66  */
    /* JADX WARN: Removed duplicated region for block: B:412:0x0c67  */
    /* JADX WARN: Removed duplicated region for block: B:415:0x0c80  */
    /* JADX WARN: Removed duplicated region for block: B:416:0x0c81 A[Catch: Exception -> 0x0c02, PHI: r3 r7 r25
      0x0c81: PHI (r3v18 long) = (r3v17 long), (r3v145 long) binds: [B:414:0x0c7e, B:20:0x0074] A[DONT_GENERATE, DONT_INLINE]
      0x0c81: PHI (r7v17 b9.r0) = (r7v16 b9.r0), (r7v81 b9.r0) binds: [B:414:0x0c7e, B:20:0x0074] A[DONT_GENERATE, DONT_INLINE]
      0x0c81: PHI (r25v18 java.util.concurrent.atomic.AtomicBoolean) = (r25v17 java.util.concurrent.atomic.AtomicBoolean), (r25v92 java.util.concurrent.atomic.AtomicBoolean) binds: [B:414:0x0c7e, B:20:0x0074] A[DONT_GENERATE, DONT_INLINE], TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:418:0x0c8a A[Catch: Exception -> 0x0c02, TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:422:0x0ca1 A[Catch: Exception -> 0x0c02, TryCatch #32 {Exception -> 0x0c02, blocks: (B:426:0x0cc4, B:416:0x0c81, B:418:0x0c8a, B:413:0x0c6c, B:408:0x0c3c, B:409:0x0c4b, B:404:0x0c22, B:395:0x0bd5, B:387:0x0bad, B:389:0x0bb1, B:391:0x0bb9, B:401:0x0c05, B:384:0x0b93, B:373:0x0b63, B:380:0x0b6f, B:422:0x0ca1, B:372:0x0b54), top: B:499:0x0b54 }] */
    /* JADX WARN: Removed duplicated region for block: B:460:0x08ed A[EXC_TOP_SPLITTER, SYNTHETIC] */
    /* JADX WARN: Removed duplicated region for block: B:494:0x09aa A[EXC_TOP_SPLITTER, SYNTHETIC] */
    /* JADX WARN: Removed duplicated region for block: B:7:0x0017  */
    /* JADX WARN: Type inference failed for: r0v243 */
    /* JADX WARN: Type inference failed for: r0v7 */
    /*
        Code decompiled incorrectly, please refer to instructions dump.
    */
```
</details>

### Jadx version

dev (built on 28 Jan 2025)

--- a/README.md
+++ b/README.md
@@ -161,6 +161,7 @@ options:
                                             'auto' - automatically select (default)
                                             'decimal' - use decimal
                                             'hexadecimal' - use hexadecimal
+  -tul, --type-update-limit              - type update limit count
   --fs-case-sensitive                    - treat filesystem as case sensitive, false by default
   --cfg                                  - save methods control flow graph to dot file
   --raw-cfg                              - save methods control flow graph (use raw instructions)

--- a/jadx-cli/src/main/java/jadx/cli/JadxCLIArgs.java
+++ b/jadx-cli/src/main/java/jadx/cli/JadxCLIArgs.java
@@ -59,6 +59,9 @@ public class JadxCLIArgs {
 	@Parameter(names = { "-j", "--threads-count" }, description = "processing threads count")
 	protected int threadsCount = JadxArgs.DEFAULT_THREADS_COUNT;
 
+	@Parameter(names = { "-tul", "--type-update-limit" }, description = "type update limit count")
+	protected int typeUpdatesLimitCount = 0;
+
 	@Parameter(names = { "--single-class" }, description = "decompile a single class, full name, raw or alias")
 	protected String singleClass = null;
 
@@ -384,6 +387,7 @@ public class JadxCLIArgs {
 		args.setFsCaseSensitive(fsCaseSensitive);
 		args.setCommentsLevel(commentsLevel);
 		args.setIntegerFormat(integerFormat);
+		args.setTypeUpdatesLimitCount(typeUpdatesLimitCount);
 		args.setUseDxInput(useDx);
 		args.setPluginOptions(pluginOptions);
 		args.setDisabledPlugins(Arrays.stream(disablePlugins.split(",")).map(String::trim).collect(Collectors.toSet()));
@@ -547,6 +551,10 @@ public class JadxCLIArgs {
 
 	public IntegerFormat getIntegerFormat() {
 		return integerFormat;
+	}
+
+	public int getTypeUpdatesLimitCount() {
+		return typeUpdatesLimitCount;
 	}
 
 	public boolean isEscapeUnicode() {

--- a/jadx-core/src/main/java/jadx/api/JadxArgs.java
+++ b/jadx-core/src/main/java/jadx/api/JadxArgs.java
@@ -167,6 +167,8 @@ public class JadxArgs implements Closeable {
 
 	private IntegerFormat integerFormat = IntegerFormat.AUTO;
 
+	private int typeUpdatesLimitCount = 0;
+
 	private boolean useDxInput = false;
 
 	public enum UseKotlinMethodsForVarNames {
@@ -738,6 +740,14 @@ public class JadxArgs implements Closeable {
 		this.integerFormat = format;
 	}
 
+	public int getTypeUpdatesLimitCount() {
+		return typeUpdatesLimitCount;
+	}
+
+	public void setTypeUpdatesLimitCount(int typeUpdatesLimitCount) {
+		this.typeUpdatesLimitCount = typeUpdatesLimitCount;
+	}
+
 	public boolean isUseDxInput() {
 		return useDxInput;
 	}
@@ -898,6 +908,7 @@ public class JadxArgs implements Closeable {
 				+ ", cfgOutput=" + cfgOutput
 				+ ", rawCFGOutput=" + rawCFGOutput
 				+ ", useHeadersForDetectResourceExtensions=" + useHeadersForDetectResourceExtensions
+				+ ", typeUpdatesLimitCount=" + typeUpdatesLimitCount
 				+ '}';
 	}
 }

--- a/jadx-core/src/main/java/jadx/core/dex/visitors/typeinference/TypeUpdate.java
+++ b/jadx-core/src/main/java/jadx/core/dex/visitors/typeinference/TypeUpdate.java
@@ -12,6 +12,7 @@ import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import jadx.api.JadxArgs;
 import jadx.core.Consts;
 import jadx.core.clsp.ClspClass;
 import jadx.core.dex.attributes.AFlag;
@@ -42,9 +43,11 @@ public final class TypeUpdate {
 	private final RootNode root;
 	private final Map<InsnType, ITypeListener> listenerRegistry;
 	private final TypeCompare comparator;
+	private final JadxArgs args;
 
 	public TypeUpdate(RootNode root) {
 		this.root = root;
+		this.args = root.getArgs();
 		this.listenerRegistry = initListenerRegistry();
 		this.comparator = new TypeCompare(root);
 	}
@@ -79,7 +82,7 @@ public final class TypeUpdate {
 			return REJECT;
 		}
 
-		TypeUpdateInfo updateInfo = new TypeUpdateInfo(mth, flags);
+		TypeUpdateInfo updateInfo = new TypeUpdateInfo(mth, flags, args);
 		TypeUpdateResult result = updateTypeChecked(updateInfo, ssaVar.getAssign(), candidateType);
 		if (result == REJECT) {
 			return result;

--- a/jadx-core/src/main/java/jadx/core/dex/visitors/typeinference/TypeUpdateInfo.java
+++ b/jadx-core/src/main/java/jadx/core/dex/visitors/typeinference/TypeUpdateInfo.java
@@ -5,6 +5,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+import jadx.api.JadxArgs;
 import jadx.core.dex.instructions.args.ArgType;
 import jadx.core.dex.instructions.args.InsnArg;
 import jadx.core.dex.nodes.MethodNode;
@@ -18,10 +19,10 @@ public class TypeUpdateInfo {
 	private final int updatesLimitCount;
 	private int updateSeq = 0;
 
-	public TypeUpdateInfo(MethodNode mth, TypeUpdateFlags flags) {
+	public TypeUpdateInfo(MethodNode mth, TypeUpdateFlags flags, JadxArgs args) {
 		this.mth = mth;
 		this.flags = flags;
-		this.updatesLimitCount = mth.getInsnsCount() * 10;
+		this.updatesLimitCount = mth.getInsnsCount() * (10 + args.getTypeUpdatesLimitCount());
 	}
 
 	public void requestUpdate(InsnArg arg, ArgType changeType) {
@@ -32,7 +33,8 @@ public class TypeUpdateInfo {
 					+ ", insn: " + arg.getParentInsn());
 		}
 		if (updateSeq > updatesLimitCount) {
-			throw new JadxOverflowException("Type inference error: updates count limit reached");
+			throw new JadxOverflowException("Type inference error: updates count limit reached"
+					+ " with updateSeq = " + updateSeq + ". Try increasing the type limit count on preferences.");
 		}
 	}
 

--- a/jadx-gui/src/main/java/jadx/gui/settings/JadxSettings.java
+++ b/jadx-gui/src/main/java/jadx/gui/settings/JadxSettings.java
@@ -740,6 +740,10 @@ public class JadxSettings extends JadxCLIArgs {
 		this.integerFormat = format;
 	}
 
+	public void setTypeUpdatesLimitCount(int typeUpdatesLimitCount) {
+		this.typeUpdatesLimitCount = typeUpdatesLimitCount;
+	}
+
 	public void setLineNumbersMode(LineNumbersMode lineNumbersMode) {
 		this.lineNumbersMode = lineNumbersMode;
 	}

--- a/jadx-gui/src/main/java/jadx/gui/settings/ui/JadxSettingsWindow.java
+++ b/jadx-gui/src/main/java/jadx/gui/settings/ui/JadxSettingsWindow.java
@@ -640,6 +640,16 @@ public class JadxSettingsWindow extends JDialog {
 			needReload();
 		});
 
+		int typeUpdatesLimitValue = settings.getTypeUpdatesLimitCount();
+		int typeUpdatesLimitCountMax = (int) Math.pow(2, 32);
+		SpinnerNumberModel typeUpdatesLimitCountSpinnerModel =
+				new SpinnerNumberModel(typeUpdatesLimitValue, 0, typeUpdatesLimitCountMax, 1);
+		JSpinner typeUpdatesLimitCount = new JSpinner(typeUpdatesLimitCountSpinnerModel);
+		typeUpdatesLimitCount.addChangeListener(e -> {
+			settings.setTypeUpdatesLimitCount((Integer) typeUpdatesLimitCount.getValue());
+			needReload();
+		});
+
 		SettingsGroup other = new SettingsGroup(NLS.str("preferences.decompile"));
 		other.addRow(NLS.str("preferences.threads"), threadsCount);
 		other.addRow(NLS.str("preferences.excludedPackages"),
@@ -664,6 +674,7 @@ public class JadxSettingsWindow extends JDialog {
 		other.addRow(NLS.str("preferences.useKotlinMethodsForVarNames"), kotlinRenameVars);
 		other.addRow(NLS.str("preferences.commentsLevel"), commentsLevel);
 		other.addRow(NLS.str("preferences.integerFormat"), integerFormat);
+		other.addRow(NLS.str("preferences.typeUpdatesCountLimit"), typeUpdatesLimitCount);
 		return other;
 	}
 

--- a/jadx-gui/src/main/resources/i18n/Messages_de_DE.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_de_DE.properties
@@ -269,6 +269,7 @@ preferences.xposed_codegen_language=Xposed-Code-Generierungssprache
 preferences.update_channel=Jadx-Updatekanal
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=Ganzzahlformat
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Schriftart ändern
 preferences.smali_font=Monospaced-Schriftart (Smali/Hex)
 preferences.laf_theme=Thema

--- a/jadx-gui/src/main/resources/i18n/Messages_en_US.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_en_US.properties
@@ -269,6 +269,7 @@ preferences.xposed_codegen_language=Xposed code generation language
 preferences.update_channel=Jadx update channel
 preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=Integer format
+preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Editor font
 preferences.smali_font=Monospaced font (Smali/Hex)
 preferences.laf_theme=Theme

--- a/jadx-gui/src/main/resources/i18n/Messages_es_ES.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_es_ES.properties
@@ -269,6 +269,7 @@ preferences.raw_cfg=Generate RAW CFG graphs
 #preferences.update_channel=Jadx update channel
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 #preferences.integerFormat=Integer format
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Fuente del editor
 #preferences.smali_font=Monospaced font (Smali/Hex)
 #preferences.laf_theme=Theme

--- a/jadx-gui/src/main/resources/i18n/Messages_id_ID.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_id_ID.properties
@@ -269,6 +269,7 @@ preferences.raw_cfg=Hasilkan grafik CFG mentah
 #preferences.update_channel=Jadx update channel
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=Format bilangan bulat
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Font editor
 preferences.smali_font=Font monospasi (Smali/Hex)
 preferences.laf_theme=Tema

--- a/jadx-gui/src/main/resources/i18n/Messages_ko_KR.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_ko_KR.properties
@@ -269,6 +269,7 @@ preferences.raw_cfg=RAW CFG 그래프 생성
 #preferences.update_channel=Jadx update channel
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 #preferences.integerFormat=Integer format
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=에디터 글씨체
 #preferences.smali_font=Monospaced font (Smali/Hex)
 preferences.laf_theme=테마

--- a/jadx-gui/src/main/resources/i18n/Messages_pt_BR.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_pt_BR.properties
@@ -269,6 +269,7 @@ preferences.raw_cfg=Gera gráficos CFG no formato RAW
 #preferences.update_channel=Jadx update channel
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 #preferences.integerFormat=Integer format
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Fonte do editor
 #preferences.smali_font=Monospaced font (Smali/Hex)
 preferences.laf_theme=Tema

--- a/jadx-gui/src/main/resources/i18n/Messages_ru_RU.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_ru_RU.properties
@@ -269,6 +269,7 @@ preferences.xposed_codegen_language=Язык генерации Xposed хуко�
 preferences.update_channel=Канал обновления Jadx
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=Формат чисел
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=Шрифт редактора Java
 preferences.smali_font=Шрифт smali/HEX редактора
 preferences.laf_theme=Тема приложения

--- a/jadx-gui/src/main/resources/i18n/Messages_zh_CN.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_zh_CN.properties
@@ -269,6 +269,7 @@ preferences.xposed_codegen_language=Xposed代码生成语言
 preferences.update_channel=Jadx 更新通道
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=数值格式化
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=编辑器字体
 preferences.smali_font=等宽字体 (Smali/Hex)
 preferences.laf_theme=主题

--- a/jadx-gui/src/main/resources/i18n/Messages_zh_TW.properties
+++ b/jadx-gui/src/main/resources/i18n/Messages_zh_TW.properties
@@ -269,6 +269,7 @@ preferences.xposed_codegen_language=Xposed 程式碼產生語言
 preferences.update_channel=Jadx 更新頻道
 #preferences.disable_tooltip_on_hover=Disable tooltip on hover
 preferences.integerFormat=整數模式
+#preferences.typeUpdatesCountLimit=Update type limit count
 preferences.font=編輯器字型
 preferences.smali_font=等寬字型 (Smali/Hex)
 preferences.laf_theme=主題

